packages/opds-web-client/src/AuthPlugin.ts

```diff
@@ -7,15 +7,15 @@ import { AuthMethod, AuthCredentials } from "./interfaces";
 /** Applications can implement this interface if they would like to support authentication
     methods other than basic auth. A list of `AuthPlugin`s can be passed as a prop to the
     `OPDSCatalog` component. */
-interface AuthPlugin {
+interface AuthPlugin<T extends AuthMethod = AuthMethod> {
```

Comment:
Is the `= AuthMethod` similar to passing a default type if the caller just calls `AuthPlugin` without a type after?

Reply:
Yep exactly